broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/AsyncSnapshotDirector.java

```diff
@@ -56,7 +58,7 @@
   private Long lastWrittenPosition;
   private TransientSnapshot pendingSnapshot;
   private long lowerBoundSnapshotPosition;
-  private boolean takingSnapshot;
+  private CompletableActorFuture<Optional<PersistedSnapshot>> snapshotFuture;
```

Comment:
:wrench: Tbh it looks a bit weird the combination of Future and Optional :sweat_smile:

Reply:
I sort of agree but couldn't find a better solution. The optional is only needed for the case that we are already taking a snapshot and thus ignoring the request for a new one, and in the case where we haven't processed anything yet.

I think both cases don't really justify completing the future exceptionally.

Reply:
I've tried out using sealed interfaces and records for the result type of the future but that made writing the assertions very cumbersome.

Reply:
Okay, I've gone back to a future of a nullable snapshot which should be a bit more readable now :+1: